plugins/scaffolder-backend/src/scaffolder/actions/builtin/publish/githubPullRequest.test.ts

```diff
@@ -0,0 +1,199 @@
+/*
+ * Copyright 2021 Spotify AB
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import fs from 'fs';
+import os from 'os';
+import path from 'path';
+import { Writable } from 'stream';
+import { v4 as uuid } from 'uuid';
+import {
+  PullRequestCreator,
+  GithubPullRequestActionInput,
+  createPublishGithubPullRequestAction,
+} from './githubPullRequest';
+import { ActionContext, TemplateAction } from '../../types';
+import { getRootLogger } from '@backstage/backend-common';
+
+import {
+  ScmIntegrationRegistry,
+  ScmIntegrations,
+} from '@backstage/integration';
+import { ConfigReader } from '@backstage/config';
+
+const createFiles = (
+  dirname: string,
+  files: Record<string, string>,
+): string => {
+  const dir = path.join(os.tmpdir(), dirname);
+  fs.mkdirSync(dir);
+
+  for (const [filePath, body] of Object.entries(files)) {
+    const fullPath = path.join(dir, filePath);
+    fs.mkdirSync(path.dirname(fullPath), { recursive: true });
+    fs.writeFileSync(fullPath, body);
+  }
+
+  return dir;
+};
+
+describe('createPublishGithubPullRequestAction', () => {
+  let instance: TemplateAction<GithubPullRequestActionInput>;
+  let fakeClient: PullRequestCreator;
+
+  let clientFactory: (
+    integrations: ScmIntegrationRegistry,
+    input: GithubPullRequestActionInput,
+  ) => Promise<PullRequestCreator>;
+
+  beforeEach(() => {
+    const integrations = ScmIntegrations.fromConfig(new ConfigReader({}));
+    fakeClient = {
+      createPullRequest: jest.fn(async (_: any) => {
+        return {
+          url: 'https://api.github.com/myorg/myrepo/pull/123',
+          headers: {},
+          status: 201,
+          data: {
+            html_url: 'https://github.com/myorg/myrepo/pull/123',
+          },
+        };
+      }),
+    };
+    clientFactory = jest.fn(async () => fakeClient);
+
+    instance = createPublishGithubPullRequestAction({
+      integrations,
+      clientFactory,
+    });
+  });
+
+  describe('with no sourcePath', () => {
+    const dirname = uuid();
+    let dir: string;
+    let input: GithubPullRequestActionInput;
+    let ctx: ActionContext<GithubPullRequestActionInput>;
+
+    beforeEach(() => {
+      input = {
+        owner: 'myorg',
+        repo: 'myrepo',
+        title: 'Create my new app',
+        branchName: 'new-app',
+        description: 'This PR is really good',
+      };
+
+      dir = createFiles(dirname, { 'file.txt': 'Hello there!' });
```

Comment:
think you can replace this `mockFiles` thing with `mock-fs` instead? It looks very similar, and you don't have to worry about polluting files and cleaning up :)

Reply:
If that's a thing, I will do with pleasure 😂